tkg/client/machine_deployment.go

```diff
@@ -103,12 +105,20 @@ func (c *TkgClient) SetMachineDeployment(options *SetMachineDeploymentOptions) e
 	if err != nil {
 		return errors.Wrap(err, "unable to determine if cluster is clusterclass based")
 	}
+
 	if ccBased {
-		var cluster capi.Cluster
-		if err = clusterClient.GetResource(&cluster, options.ClusterName, options.Namespace, nil, nil); err != nil {
+		cluster := &capi.Cluster{}
+		if err := clusterClient.GetResource(cluster, options.ClusterName, options.Namespace, nil, nil); err != nil {
 			return errors.Wrap(err, "Unable to retrieve cluster resource")
 		}
-		return DoSetMachineDeploymentCC(clusterClient, &cluster, options)
+		skip, err := skipMDCreation(clusterClient, c, cluster, options)
+		if err != nil {
+			return err
+		}
+		if skip {
+			return nil
+		}
```

Comment:
Question: Is this required because the clusterclass will have MD? I am asking because the cluster manifest will not have workers spec.. no? Also, could the skip logic be derived from some annotation on the cluster object?

Reply:
In case of single node cluster, we don't need MD creation as there are no worker nodes.
But in case of standard cluster, MD creation is needed, so this func just checks if the MD creation is to be skipped or not.

Reply:
> Also, could the skip logic be derived from some annotation on the cluster object?

We are not introducing any annotations for single node cluster, and we have access to machinedeployments only, at this point, so I guess we dont have any other alternative to derive skip logic, but to depend on topology.

Reply:
Is there any reason a single node cluster should be disallowed from adding node pools?